### ISSUE TITLE
In progress bar distinguish initial delay from rest? 

### ISSUE THREAD

**Issue by rocky** [ConsenSys/truffle-security#95]
The progress bars are fantastic Something that might make them even more awesome would be to distinguish the part where we are in the inital delay versus the part after that. One possibility is another color, another is a different symbol than '*'. 

This way people can gauge whether and how to adjust the initial delay.  Thoughts @tagomaru ?

**Comment by tagomaru** [comment]
@rocky 
OK, I will think about it, however I am not clear about `initialDelay` yet.

According to [Improving Polling Response](https://github.com/ConsenSys/armlet#improving-polling-response), it is a parameter that armlet waits before attempting its first status pool.

However I could not find whether this parameter affects to it...

https://github.com/ConsenSys/armlet/blob/master/index.js#L118

If the above is false, as a next step, polling seems to start at the below.

https://github.com/ConsenSys/armlet/blob/master/index.js#L128

I expected that it waits until initialDelay time passes before the below step, but there is no wating.

https://github.com/ConsenSys/armlet/blob/master/lib/analysisPoller.js#L177

Could I ask you where and how the initialDelay works as [Improving Polling Response](https://github.com/ConsenSys/armlet#improving-polling-response) says?

**Comment by rocky** [comment]
We want to encourage good guessing of initial delay. And in IDE's we probably will be using historical data to do so. 

The following is from `README.md`. Does this help? 

There are two time parameters, given in milliseconds, that change how quickly a analysis result is reported back:

* initial delay
* maximum delay

The initial delay is the minimum amount of time that this library waits before attempting its first status poll. 

Note however that if a request has been cached, then results come back immediately and no status polling is needed.  (The server caches previous analysis runs; it takes into account the data passed to it, the analysis mode, and the back-end versions of components used to provide the analysis.)

The maximum delay is the maximum amount of time we will wait for an analysis to complete. Note, however, that if the processing has not finished when this timeout is reached, it may still be running on the server side. Therefore when a timeout occurs, you will get back a UUID which can subsequently be used to get status and results.

The closer these two parameters are to the actual time range that is needed by analysis, the faster the response will get reported back after completion on the server end.

**Comment by rocky** [comment]
The followiing used to be in README.md but was removed because it was felt that this was too verbose: 

Until we have a websocket interface so the server can directlypass back results without any additional action required on the server side, your REST API requires the client to poll for status. We have
seen that this polling can cause a lot of overhead, if not done judiciously. So, each request is allowed up to 10 status probes.

We have seen that _no_ analysis request will finish in less than a certain period of time. Since the number of probe per analysis is limited, it doesn't make sense to probe before the fastest
analysis-completion time.

The 10 status probes are done in geometrically increasing time intervals. The first interval is the shortest and the last interval is the longest. The response rate at the beginning is better than the response rate at the end, in terms of how much additional time it takes before the analysis completion is noticed.

However this progression is not fixed. Instead, it takes into account the maximum amount of time you are willing to wait for a result.

_In other words, the shorter the short period of time you give for the maximum timeout, the shorter the geometric succession of the 10 probes allotted to an analysis request will be._

To make this clear, if you only want to wait a maximum of two minutes, then the first delay will be 0.3 seconds, while the delay before last poll will be about half a minute. If on the other hand you want to wait up to 2 hours, then the first delay will be 9 seconds, and the last one will be about 15 minutes.

**Comment by tagomaru** [comment]
@rocky 
> The initial delay is the minimum amount of time that this library waits before attempting its first status poll.

Sorry, but still strange for me.

According to [this](https://github.com/ConsenSys/armlet/blob/master/lib/analysisPoller.js#L167-L189), do you mean that initialDelay parameter is not for first waiting time itself but for affecting parameter to first waiting times?
(it also affects all of the waiting times, though)

**Comment by rocky** [comment]
That code occurs only after we discover the result is not cached and have waited for the initialDelay has been waited for. 

Suppose you have 6 polling opportunites that look like this: 

```
!  !    !    !     !       !               !
```

And you want to minimize the number of polls and the number of delay after a MythX job completes? 
Suppose based on past history the completion time is in this range: 

```
                          =============
```

If you fit the two together you would get: 

```
!  !    !    !     !       !               !
                          ============
```
If instead you added an initial delay (to the polls the white space before the first `!`) you would get: 

```
                    !  !    !    !     !       !               !
```
                          ============

**Comment by tagomaru** [comment]
@rocky 
Yeah, I understood that, but what i cannot understand is  

> have waited for the initialDelay has been waited for.

Which code works for this?

**Comment by tagomaru** [comment]
Accoding to [this](https://github.com/ConsenSys/armlet/blob/master/index.js#L126-L128),
there seems to be no wating during initialDealy.

**Comment by rocky** [comment]
Ah - yes you are correct - that is a bug! there should be an initial delay done in analysisPoller.

**Comment by tagomaru** [comment]
Oh,,, will i fix it?

**Comment by tagomaru** [comment]
By the way, if this is fixed, the number of polling except fist one should be changed from 10 to 9?
The image is like below.

1. First waiting time is initialDelay.
2. The rest waiting times are calculated base on [this](https://www.wolframalpha.com/input/?i=n+%3D+sum+(x*i)**2,++i%3D1+to+9+solve+for+x).

I thinks sqrt(385) should be changed to sqrt(285).

**Comment by rocky** [comment]
> Oh,,, will i fix it?

If you would - then please do. 

The 385 is for 10 and then 285 is for 9.  There is nothing magic about either 9+1  or 10+1.  I had thought for analysis in associating maxumum number of polls per analysis request 10 might be an easy number and it looks like right now I have 11. So sure if in  that you want to go with 9 and 285, that's fine.

**Comment by tagomaru** [comment]
@rocky 
OK, thanks for your comment.
I will think about it and fix it soon.

**Comment by tagomaru** [comment]
OK, i will work on it at a later date.
Howerver, [by this code](https://github.com/ConsenSys/armlet/blob/master/index.js#L133-L145), it may not make sense if refresh token occurs, since it waits again during intial delay time.

**Comment by rocky** [comment]
So that's a bug that we should fix. Note however that after the refresh is done there won't be another one for a day. So as a problem, it is not a big one.

**Comment by tagomaru** [comment]
Yes. anyway after i work on this, users can notice whehter initial delay passed or not, which means users can notice whether the first response status is 'Finished' or not.

**Comment by nbanmp** [comment]
Closing as this is very low priority. If anyone believes this is worth implementing, please reopen and explain.
